# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: devtools/server/actors/watcher.js
# Commit: 895cf08432c4
# Full Hash: 895cf08432c40fe371ff83e846cb161430c6be4b
# Author: Nicolas Chevobbe <nchevobbe@mozilla.com>
# Date: 2020-10-27 04:41:26
# Regressor Bug: 1633712
# File Overlap Count: 1
# Description:
#   Bug 1633712 - [devtools] Create WorkerTargets as soon as possible. r=ochameau,devtools-backward-compat-reviewers.
#   
#   This patch adds support for dedicated worker targets in the Watcher actor.
#   Shared and Service workers are not handled yet.
#   
# ==============================================================================

diff -r 2ce79a9ed2d5 -r 895cf08432c4 devtools/server/actors/watcher.js
--- a/devtools/server/actors/watcher.js	Mon Oct 26 12:52:26 2020 +0000
+++ b/devtools/server/actors/watcher.js	Mon Oct 26 12:53:43 2020 +0000
@@ -27,6 +27,11 @@
   Targets.TYPES.PROCESS,
   "devtools/server/actors/watcher/target-helpers/process-helper"
 );
+loader.lazyRequireGetter(
+  TARGET_HELPERS,
+  Targets.TYPES.WORKER,
+  "devtools/server/actors/watcher/target-helpers/worker-helper"
+);
 
 exports.WatcherActor = protocol.ActorClassWithSpec(watcherSpec, {
   /**
@@ -99,8 +104,10 @@
       traits: {
         // FF77+ supports frames in Watcher actor
         [Targets.TYPES.FRAME]: true,
-        // FF83+ supports processes
+        // FF84+ supports processes
         [Targets.TYPES.PROCESS]: true,
+        // FF84+ supports workers in Watcher actor for content toolbox.
+        [Targets.TYPES.WORKER]: hasBrowserElement,
         resources: {
           // FF81+ (bug 1642295) added support for:
           // - CONSOLE_MESSAGE